src/hdmf_zarr/backend.py

```diff
@@ -546,14 +547,41 @@ def write_group(self, **kwargs):
                     export_source=export_source,
                 )
 
-        # write all links (haven implemented)
         links = builder.links
         if links:
+            # Note: In HDMF, we simply call write_link (within write_group) which contains a similar logic as seen
+            # below. This is not the case here (until a refactor).
```

Comment:
It looks like the only place where `write_link` is being called is in this for loop. Why not move the logic to `write_link` directly now? I think that would help keep the code much more readable as it keeps the link logic in one place.